### PR TITLE
More plugins

### PR DESCRIPTION
**Pull request by tkstanczak**

### CODE REVIEW

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Network/P2P/SyncPeerProtocolHandlerBase.cs [hunk 1]
```diff
@@ -41,11 +39,71 @@
 
 namespace Nethermind.Network.P2P
 {
+    public class MessageQueue<TMsg, TData> where TMsg : MessageBase
```
[the remark this answers is not in the document]
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Network/P2P/SyncPeerProtocolHandlerBase.cs:42]: ok

File: src/Nethermind/Nethermind.Network/P2P/SyncPeerProtocolHandlerBase.cs [hunk 2]
```diff
@@ -41,11 +39,71 @@
...
+
+                _currentRequest.ResponseSize = size;
+                _currentRequest.CompletionSource.SetResult(data);
+                if (_requestQueue.TryDequeue(out _currentRequest))
+                {
+                    _currentRequest.StartMeasuringTime();
+                    _send(_currentRequest.Message);
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Network/P2P/SyncPeerProtocolHandlerBase.cs:94]: only one lock so deadlock not possible

File: src/Nethermind/Nethermind.Blockchain/Visitors/HeaderVisitOutcome.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using System;
+
+namespace Nethermind.Blockchain.Visitors
+{
+    [Flags]
+    public enum HeaderVisitOutcome
+    {
+        None,
+        StopVisiting = 1,
+        All = 1
```
[the remark this answers is not in the document]
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Blockchain/Visitors/HeaderVisitOutcome.cs:27]: only one flag but more consistent with other types which have more than one flag and similar naming

File: src/Nethermind/Nethermind.Network/P2P/SyncPeerProtocolHandlerBase.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Network/P2P/SyncPeerProtocolHandlerBase.cs]: 2 locks one in Send one in Handle

[the remark this answers is not in the document]
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Network/P2P/SyncPeerProtocolHandlerBase.cs]: on the same object so deadlock not possible

**Pull request merged**